### PR TITLE
Support error messages in plain text.

### PR DESCRIPTION
**Pull request by monaka**
### What does this PR do?

Supports error messages in plain text.

It might be better it is fixed in the server side also. But I fixed the client side only in this PR.

### What issues does this PR fix or reference?

None.

### CODE REVIEW

File: ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html [hunk 1]
```diff
@@ -186,7 +186,13 @@
                         return;
                     }
                     if (this.status !== 200) {
-                        var response = JSON.parse(this.responseText);
+                        var response;
```
Comment (sleshchenko) [inline ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html]:
```suggestion
                        var message;
```

File: ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html [hunk 2]
```diff
@@ -186,7 +186,13 @@
...
+                        try {
+                            response = JSON.parse(this.responseText);
```
Comment (sleshchenko) [inline ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html]:
```suggestion
                            message = JSON.parse(this.responseText).message;
```

File: ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html [hunk 3]
```diff
@@ -186,7 +186,13 @@
...
+                        } catch(e) {
+                            /* Maybe plain text. */
+                            response = this.responseText;
```
Comment (sleshchenko) [inline ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html]:
```suggestion
                            message = this.responseText;
```

File: ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html [hunk 4]
```diff
@@ -186,7 +186,13 @@
...
+                        }
                         errorMessage = 'Failed to get the workspace: "' + response.message;
```
Comment (sleshchenko) [inline ide/che-ide-gwt-app/src/main/resources/org/eclipse/che/ide/public/IDE.html]:
```suggestion
                        errorMessage = 'Failed to get the workspace: "' + message;
```

**Review by sleshchenko — COMMENTED** [review]
Looking forward when Github will introduce segments suggestions.

Did you check your changes? Won't `response.message` be empty/null if response is plain text?

**Comment by monaka** [comment]
> Won't response.message be empty/null if response is plain text?

Will be always `undefined`. It's my mistake and thanks for your help. I applied your fix.

**Review by vparfonov — APPROVED** [review]

**Comment by musienko-maxim** [comment]
Can one of the admins verify this PR?

**Comment by monaka** [comment]
@musienko-maxim This PR is blocked by `ci-build-check` and I have no permission to run it.

**Comment by monaka** [comment]
@skabashnyuk I reviewed ci-test result. It seems this PR branch didn't have PR #13176 (merged 4 days ago). I updated to the master branch and force-pushed.

Screenshot 2019-04-23 at 12 23 29

**Comment by monaka** [comment]
Could somebody run `ci-build` ?  It seems I have no permission to it.

**Comment by sleshchenko** [comment]
ci-build

**Pull request merged**